### ISSUE TITLE
Line display issues

### ISSUE THREAD

**Issue by LiHRaM** [xi-editor/xi-win#44]
# Summary
xi-win _sometimes_ seems to have trouble displaying more than 12 contiguous lines
1. File save/open
2. Using the mouse to move the cursor
3. Selections

# Steps to reproduce
1. Open file with more than 12 lines
2. Hold `shift + down arrow` keys and watch the fireworks

There seem to be a lot of ways you can make lines disappear. I've noticed that sometimes when I click on a line with the mouse, an arbitrary line elsewhere in the file will go blank.

**Comment by raphlinus** [comment]
Yes, definitely a problem. After playing with it a bit, I think this is because `update_viewport` is often not being called when it should, so core has an inaccurate model of the viewport and thus fails to keep visible lines up to date. At the least, a call to `EditView::size` should update the viewport.

As partial explanation, this probably got broken when the EditView was reimplemented as a widget for xi-win-ui, rather than directly on top of the shell. But I haven't checked the code history; it's possible it never worked correctly, though I do see efforts to update the viewport on, for example, scrolling through mousewheel and ctrl-up/down.

Thanks for looking into it!
